packages/nextjs/src/config/webpack.ts

```diff
@@ -341,9 +341,11 @@ export function constructWebpackConfigFunction(
       newConfig.entry = async () => addSentryToClientEntryProperty(origEntryProperty, buildContext);
     }
 
-    // We don't want to do any webpack plugin stuff OR any source maps stuff in dev mode.
+    const isStaticExport = userNextConfig?.output === 'export';
+
+    // We don't want to do any webpack plugin stuff OR any source maps stuff in dev mode or for the server on static-only builds.
     // Symbolication for dev-mode errors is done elsewhere.
-    if (!isDev) {
+    if (!isDev || !(isStaticExport && isServer)) {
```

Comment:
```suggestion
    if (!isDev && !(isStaticExport && isServer)) {
```

Only handle webpack when we're in prod AND we don't have a `output: 'export'` build